submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,429 @@
+    },
+}
+
+function findSubstring(string, substring) {
+    return string.toLowerCase().includes(substring.toLowerCase());
+}
+
+function findMatchesWithPropertiesValues(
+    propertiesList,
+    friendsArray,
+    substring
+) {
+    return friendsArray.filter((friend) => {
+        return propertiesList.filter((property) => findSubstring(friend[property], substring)).length;
+    });
+}
+
+function filterByGender(e, friendsArray) {
+    let checkboxes = Array.from(document.querySelectorAll(".checkbox:checked"));
+    return friendsArray.filter((friend) =>
+        checkboxes.some((gender) => friend.gender === gender.value)
+    );
+}
+
+function filterByAge(friendsArray) {
+    if (MIN_AGE_INPUT.value) friendsArray = friendsArray
+        .filter((friend) => friend.age >= MIN_AGE_INPUT.value);
+    if (MAX_AGE_INPUT.value) friendsArray = friendsArray
+        .filter((friend) => friend.age <= MAX_AGE_INPUT.value);
+    return friendsArray;
+}
+
+function checkFiltersChanged(event) {
+    return ["list__label", "select__container", "number", "checkbox"]
+        .some((filter) => {
+            if (!(event.type === "click" &&  event.target.classList.contains("select__container"))) {
+                return event.target.classList.contains(filter);
+            }
+        });
+}
+
+function updateCards(event) {
+    if (checkFiltersChanged(event)) {
+        FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY;
+        sortCards(event, FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByAge(FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByGender(event, FRIENDS_ARRAY);
+        appendFriendsCards(FRIENDS_ARRAY);
+    }
+}
+
+function isSelectModified() {
+    return SELECT_CONTAINER.attributes["select-modified"].value === "true";
+}
+
+function checkOptionsVisibility() {
+    return OPTIONS_CONTAINER.classList.contains("visible");
+}
+
+function resetPreviouslySelectedOptions(selectedOption) {
+    Array.from(OPTIONS_LIST).forEach((option) => {
+        if (option.textContent != selectedOption) {
+            document.getElementById(option.htmlFor).checked = false;
+        }
+    });
+}
+
+function removeAriaSelectedAttribute(option) {
+    option.removeAttribute("aria-selected");
+}
+
+function getSelectOption(selectText) {
+    return Array.from(OPTIONS_LIST).find(
+        (option) => option.textContent === selectText
+    );
+}
+
+function higlightSelectedOption(itemToSelect) {
+    itemToSelect.setAttribute("aria-selected", "true");
+    document.getElementById(itemToSelect.htmlFor).checked = true;
+    updateSelectText(itemToSelect);
+}
+
+function focusOnItem(buttonPressed) {
+    const optionsListArray = Array.from(OPTIONS_LIST),
+        selectedItem = optionsListArray.find(
+            (listItem) => listItem.textContent === selectedOption
+        );
+    let selectedItemIndex = Array.from(OPTIONS_LIST).indexOf(selectedItem);
+    buttonPressed === "ArrowUp" ? selectedItemIndex-- : selectedItemIndex++;
```

Comment:
Don't use ternary operators in such a way :) Replace it with `if ... else` statement